activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -367,7 +367,9 @@ def convert_key(key) # :doc:
         key.kind_of?(Symbol) ? key.to_s : key
       end
 
-      def convert_value(value, options = {}) # :doc:
+      EMPTY_HASH = {}.freeze
+
+      def convert_value(value, options = EMPTY_HASH) # :doc:
```

Comment:
How about changing to `for: nil`? `:for` is the only valid option.

Reply:
Converting over to use keyword args with a default value sounds great to me. A few tests show MRI 2.4, 2.5 and 2.6 are clever enough to avoid allocating memory each time the function is invoked using the default value. `for` is a keyword in Ruby so I'll have to use a different parameter name. Maybe `for_usage`?

Reply:
hmmm, we can't rename the option because this method is public API. Just not sure if this `for` option was. I guess we will have to live with this patch.